docker_test.go

```diff
@@ -96,46 +101,48 @@ func TestContainerPushBuildNameNumberFromEnv(t *testing.T) {
 	}
 }
 
 func TestContainerPull(t *testing.T) {
 	containerManagers := initContainerTest(t)
-	for _, containerManager := range containerManagers {
-		imageTag := inttestutils.BuildTestContainerImage(t, tests.DockerImageName, containerManager)
+	for _, repo := range []string{*tests.DockerVirtualRepo, *tests.DockerTargetRepo} {
+		for _, containerManager := range containerManagers {
+			imageTag := inttestutils.BuildTestContainerImage(t, tests.DockerImageName, containerManager)
 
-		// Push container image
-		artifactoryCli.Exec(containerManager.String()+"-push", imageTag, *tests.DockerTargetRepo)
+			// Push container image
+			artifactoryCli.Exec(containerManager.String()+"-push", imageTag, repo)
 
-		buildNumber := "1"
+			buildNumber := "1"
 
-		// Pull container image
-		artifactoryCli.Exec(containerManager.String()+"-pull", imageTag, *tests.DockerTargetRepo, "--build-name="+tests.DockerBuildName, "--build-number="+buildNumber)
-		artifactoryCli.Exec("build-publish", tests.DockerBuildName, buildNumber)
+			// Pull container image
+			artifactoryCli.Exec(containerManager.String()+"-pull", imageTag, repo, "--build-name="+tests.DockerBuildName, "--build-number="+buildNumber)
+			artifactoryCli.Exec("build-publish", tests.DockerBuildName, buildNumber)
 
-		imagePath := path.Join(*tests.DockerTargetRepo, tests.DockerImageName, "1") + "/"
-		validateContainerBuild(tests.DockerBuildName, buildNumber, imagePath, tests.DockerImageName+":1", 0, 7, 7, t)
+			imagePath := path.Join(repo, tests.DockerImageName, "1") + "/"
+			validateContainerBuild(tests.DockerBuildName, buildNumber, imagePath, tests.DockerImageName+":1", 0, 7, 7, t)
 
-		buildNumber = "2"
-		artifactoryCli.Exec(containerManager.String()+"-pull", imageTag, *tests.DockerTargetRepo, "--build-name="+tests.DockerBuildName, "--build-number="+buildNumber, "--module="+ModuleNameJFrogTest)
-		artifactoryCli.Exec("build-publish", tests.DockerBuildName, buildNumber)
-		validateContainerBuild(tests.DockerBuildName, buildNumber, imagePath, ModuleNameJFrogTest, 0, 7, 7, t)
+			buildNumber = "2"
+			artifactoryCli.Exec(containerManager.String()+"-pull", imageTag, repo, "--build-name="+tests.DockerBuildName, "--build-number="+buildNumber, "--module="+ModuleNameJFrogTest)
+			artifactoryCli.Exec("build-publish", tests.DockerBuildName, buildNumber)
+			validateContainerBuild(tests.DockerBuildName, buildNumber, imagePath, ModuleNameJFrogTest, 0, 7, 7, t)
 
-		inttestutils.ContainerTestCleanup(t, artifactoryDetails, artHttpDetails, tests.DockerImageName, tests.DockerBuildName)
+			inttestutils.ContainerTestCleanup(t, artifactoryDetails, artHttpDetails, tests.DockerImageName, tests.DockerBuildName, repo)
+		}
 	}
 }
 
-func containerTestCleanup(t *testing.T, imageName, buildName string) {
+func containerTestCleanup(t *testing.T, imageName, buildName, repo string) {
```

Comment:
There are a f ew functions in this file, which publish build-info, but instead of invoking this function, it invokes ```inttestutils.ContainerTestCleanup``` directly, and the therefore, the published build doesn't get deleted.

Reply:
No calls are made to this function (probably old).
This function `inttestutils.ContainerTestCleanup` is getting called by all docker tests and it includes the deletion of the published build.